source/features/merged-pr-first-published-tag.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+import select from 'select-dom';
+import TagIcon from 'octicon/tag.svg';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../libs/features';
+import fetchDom from '../libs/fetch-dom';
+import {getRepoURL} from '../libs/utils';
+import observeElement from '../libs/simplified-element-observer';
+
+const getFirstTag = cache.function(async (commit: string): Promise<string | undefined> => {
+	const firstTag = await fetchDom<HTMLAnchorElement>(`/${getRepoURL()}/branch_commits/${commit}`, 'ul.branches-tag-list li:last-child a');
+
+	return firstTag?.textContent as string;
```

Comment:
`as string` shouldn't be necessary. Without it, the type should already be `string | undefined`

Reply:
But with the optional chaining...
```
Type 'string | null | undefined' is not assignable to type 'string | undefined'.
  Type 'null' is not assignable to type 'string | undefined'.ts(2322)
```

And if I add a `!` ex `return firstTag?.textContent!;` 
```
Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong. (@typescript-eslint/no-non-null-asserted-optional-chain)XO(@typescript-eslint/no-non-null-asserted-optional-chain)
```

Reply:
Ugh, DOM types, `null` and TypeScript all in one place.

Related: https://github.com/typescript-eslint/typescript-eslint/issues/1976

In this case, I think this should work without using assertions:

```suggestion
	return firstTag?.textContent ?? undefined;
```